app/src/androidTest/java/com/github/multimatum_team/multimatum/QRCodeReaderActivityTest.kt

```diff
@@ -0,0 +1,24 @@
+package com.github.multimatum_team.multimatum
+
+import androidx.test.core.app.ActivityScenario
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.rule.GrantPermissionRule
+import org.junit.Rule
+import org.junit.Test
+
+class QRCodeReaderActivityTest {
+    @get:Rule
+    val permissionRule: GrantPermissionRule = GrantPermissionRule.grant(
+        android.Manifest.permission.CAMERA
+    )
+
+    @Test
+    fun should_display_scanner_when_permissionsAreGranted() {
```

Comment:
You can use this syntax to explain the test scenario in plain english:
```kotlin
fun `Should display scanner when permissions are granted`() {
    ...
}
```

Reply:
@felko maybe be careful with this "plain english" syntax. I have already had problems with spaces not recognized and replaced by codes in error messages in this syntax, which is unpleasant to read. I would leave it with "_" (but it is a detail).

Reply:
For now I will let it like that. Maybe change in the future